lib/Dialect/DC/Transforms/Analysis.cpp

```diff
@@ -0,0 +1,238 @@
+//===- Analysis.cpp - Analysis Pass -----------------------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Contains the definitions of the Analysis pass.
+//
+//===----------------------------------------------------------------------===//
+#include "circt/Dialect/DC/DCOps.h"
+#include "circt/Dialect/DC/DCPasses.h"
+#include "circt/Dialect/HW/HWDialect.h"
+#include "mlir/Pass/Pass.h"
+
+#include "circt/Dialect/Comb/CombDialect.h"
+#include "circt/Dialect/Comb/CombOps.h"
+#include "circt/Dialect/DC/DCOps.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Support/IndentedOstream.h"
+#include "mlir/Tools/mlir-translate/Translation.h"
+#include "llvm/ADT/ScopedHashTable.h"
+#include "llvm/ADT/StringRef.h"
+#include "llvm/Support/raw_ostream.h"
+#include <string>
+
+namespace circt {
+namespace dc {
+#define GEN_PASS_DEF_DCDOTPRINT
+#include "circt/Dialect/DC/DCPasses.h.inc"
+} // namespace dc
+} // namespace circt
+
+using namespace circt;
+using namespace dc;
+using namespace mlir;
+
+using ValueMap = llvm::ScopedHashTable<mlir::Value, std::string>;
+
+// stores operand and results for each node in the dot graph
+struct DotNode {
+  std::string nodeType;
```

Comment:
Nit: It might be a bit nicer and more efficient to use an enum for this instead. This then ofc. requires some stringify function.

Reply:
That's what I had in mind, yes.